### PR TITLE
Refactor fixed fee calculations

### PR DESCRIPTION
**Pull request by jsugarman**
#### What

Refactor AGFS (scheme 9/10) fixed fee calculator integration

#### Why

leftover work from previous implementation to improve code quality, reduce API calls, facilitate future extendability and increase test coverage.

#### Ticket

[Refactor AGFS fixed fee calculator integration](https://dsdmoj.atlassian.net/browse/CBO-334)

#### TODO
- [x] refactoring of backend services
- [x] use fee calc API’s price endpoint to improve performance
- [X] unit test fee calculator ajax (jasmine)?? - going to rely on cucumber instead
- [x] review feature tests for fixed fee calculations
- [x] add feature tests for fixed fee calculations advocate category and parent quantity impact

### CODE REVIEW

File: features/claims/advocate/scheme_nine/advocate_fixed_fee_claim_submit.feature [hunk 1]
```diff
@@ -21,7 +21,7 @@ Feature: Advocate submits a claim for a Fixed fee (Appeal against sentence)
 
     Then I click "Continue" in the claim form
 
-    Given I insert the VCR cassette 'features/claims/advocate/scheme_nine/fixed_fee_calculations'
+    Given I insert the VCR cassette 'features/claims/advocate/scheme_nine/fixed_fee_calculations' and record 'new_episodes'
```
Comment (jsugarman) [inline features/claims/advocate/scheme_nine/advocate_fixed_fee_claim_submit.feature]: remove record new episodes

File: features/claims/advocate/scheme_ten/advocate_fixed_fee_claim_submit.feature [hunk 2]
```diff
@@ -21,7 +21,7 @@ Feature: Advocate submits a claim for a Fixed fee (Appeal against conviction)
 
     Then I click "Continue" in the claim form
 
-    Given I insert the VCR cassette 'features/claims/advocate/scheme_10/fixed_fee_calculations'
+    Given I insert the VCR cassette 'features/claims/advocate/scheme_ten/fixed_fee_calculations' and record 'new_episodes'
```
Comment (jsugarman) [inline features/claims/advocate/scheme_ten/advocate_fixed_fee_claim_submit.feature]: remove and record new episodes

File: features/fee_calculator/fixed_fee_calculator.feature [hunk 3]
```diff
@@ -21,46 +21,22 @@ Feature: Advocate requests calculator
...
     And I add a fixed fee 'Number of cases uplift' with case numbers
     Then the last fixed fee case numbers section should be visible
-    Then the last fixed fee rate should be populated with '21.60'
+    Then the last fixed fee rate should be populated with '26.00'
 
-    And I add a fixed fee of defendants uplift
+    And I add a fixed fee 'Number of defendants uplift'
     Then the last fixed fee case numbers section should not be visible
-    Then the last fixed fee rate should be populated with '21.60'
+    Then the last fixed fee rate should be populated with '26.00'
```
Comment (jsugarman) [inline features/fee_calculator/fixed_fee_calculator.feature]: Add steps to exercise 
1. changing a parent quantity impact on uplift unit price
2. changing the advocate category recalculates all prices

**Review by jsugarman — COMMENTED** [review]
self review

File: README.md [hunk 4]
```diff
@@ -117,20 +117,39 @@ To execute cucumber test scenarios
...
+```bash
+# run specs requiring LAA fee calculator API call stubs/cassettes
+$ rspec --tag fee_calc_vcr
+
+```
+
+Changes to the calling of the LAA fee calculator API will most likely require you to rerecord the VCR cassetes that stub these calls. To rerecord
+VCR cassettes you can delete the existing ones (in `vcr/cassettes/spec`). They will be recreated when the specs are run.
+
+It is a good idea to do this when changes occur to the LAA fee calculato API too.
```
Comment (colinbruce) [inline README.md]: [**not a blocker**] calculato -> calculator

File: README.md [hunk 5]
```diff
@@ -117,20 +117,39 @@ To execute cucumber test scenarios
...
+
+##### Cucumber
+Some cucumber features require VCR cassettes to stub calls to the LAA fee caclculator API. These features require and are tagged with a `@fee_calc_vcr` tag. To re-record the cassettes delete the existing ones and run the feature again. See [Create a new VCR cassette](#new_cassette) below for more.
```
Comment (colinbruce) [inline README.md]: You can hyperlink all headings in markdown by downcasing and hyphenating the header...
(#create-new-vcr-cassette)

File: README.md [hunk 6]
```diff
@@ -117,20 +117,39 @@ To execute cucumber test scenarios
...
+
+#### Internal API
+Some cucumber feature tests use VCR to record/store mock results the internal API calls (calling our own API) for certain endpoints (case worker claims in particular).
+To create a new feature/scenario that relies on such endpoints you will therefore need to record a new "cassette", as below.
+
+<a id='new_cassette'></a>
```
Comment (colinbruce) [inline README.md]: not needed for hyperlinking... see separate comment

File: README.md [hunk 7]
```diff
@@ -117,20 +117,39 @@ To execute cucumber test scenarios
...
+##### Create new VCR cassette
 
 Run this in a new console:
 
 ```bash
-RAILS_ENV=test rails s -p 3001 -P /tmp/rails3001.pid
+# Start internal API for use by test suite
+$ RAILS_ENV=test rails s -p 3001 -P /tmp/rails3001.pid
```
Comment (colinbruce) [inline README.md:151]: 👍

**Review by colinbruce — APPROVED** [review]
A couple of minor observations, but otherwise LGTM 👍

File: README.md (hunk 5 above)
Comment (jsugarman) [reply README.md]: cool, was looking for best way to do this 👍

File: README.md (hunk 6 above)
Comment (jsugarman) [reply README.md]: can't get this working in my local markdown viewer (perhaps it is github specific markdown??) or i have it wrong:

heading:
```
##### Create new VCR cassette
```

link: 
```
[Create a new VCR cassette](#create-new-vcr-cassette)
```

**Pull request merged**